ProcessLib/TwoPhaseFlowWithPrho/TwoPhaseFlowWithPrhoLocalAssembler.h

```diff
@@ -0,0 +1,191 @@
+                                            IntegrationMethod, GlobalDim>(
+              element, is_axially_symmetric, _integration_method)),
+          _process_data(process_data),
+          _saturation(
+              std::vector<double>(_integration_method.getNumberOfPoints())),
+          _pressure_nonwetting(
+              std::vector<double>(_integration_method.getNumberOfPoints()))
+    {
+        unsigned const n_integration_points =
+            _integration_method.getNumberOfPoints();
+        _ip_data.reserve(n_integration_points);
+        for (unsigned ip = 0; ip < n_integration_points; ip++)
+        {
+            _ip_data.emplace_back(*_process_data._material);
+            _ip_data[ip]._sw = 1.0;
+            _ip_data[ip]._rho_m = 0.0;
+            _ip_data[ip]._dsw_dpg = 0.0;
+            _ip_data[ip]._dsw_drho = 0.0;
+            _ip_data[ip]._drhom_dpg = 0.0;
+            _ip_data[ip]._drhom_drho = 0.0;
+        }
+    }
+
+    void assemble(double const t, std::vector<double> const& local_x,
+                  std::vector<double>& local_M_data,
+                  std::vector<double>& local_K_data,
+                  std::vector<double>& local_b_data) override;
+    void preTimestepConcrete(std::vector<double> const& /*local_x*/,
+                             double const /*t*/,
+                             double const /*delta_t*/) override
+    {
+        unsigned const n_integration_points =
+            _integration_method.getNumberOfPoints();
+
+        for (unsigned ip = 0; ip < n_integration_points; ip++)
+        {
+            _ip_data[ip].pushBackState();
+        }
+    }
+    Eigen::Map<const Eigen::RowVectorXd> getShapeMatrix(
+        const unsigned integration_point) const override
+    {
+        auto const& N = _shape_matrices[integration_point].N;
+
+        // assumes N is stored contiguously in memory
+        return Eigen::Map<const Eigen::RowVectorXd>(N.data(), N.size());
+    }
+
+    std::vector<double> const& getIntPtSaturation(
+        std::vector<double>& /*cache*/) const override
+    {
+        assert(_saturation.size() > 0);
+        return _saturation;
+    }
+
+    std::vector<double> const& getIntPtNonWettingPressure(
+        std::vector<double>& /*cache*/) const override
+    {
+        assert(_pressure_nonwetting.size() > 0);
+        return _pressure_nonwetting;
+    }
+
+private:
+    MeshLib::Element const& _element;
+
+    IntegrationMethod const _integration_method;
+    std::vector<ShapeMatrices> _shape_matrices;
+
+    TwoPhaseFlowWithPrhoProcessData const& _process_data;
+    std::vector<IntegrationPointData> _ip_data;
+    const double molar_mass_h2o =
+        MaterialLib::PhysicalConstant::MolarMass::Water;
+    const double molar_mass_h2 = MaterialLib::PhysicalConstant::MolarMass::H2;
+    // Note: currently only isothermal case is considered, so the temperature is
+    // assumed to be const
+    // the variation of temperature will be taken into account in future
```

Comment:
Good comment. Please add the `\TODO` mark here and make the comment a doxygen comment (use `///`).